main.c

```diff
@@ -969,6 +978,38 @@ int __attribute__((used)) main(void) {
     alarm_reset();
     #endif
 
+    // Pystack variables have to exist even in stackless for the function calls
```

Comment:
getenv.c should be usable from pure C code. I think it only depends on Python for throwing exceptions. The exception handling code can be moved to shared-bindings/os, then shared-module/os/getenv.c can be used from pure C. This avoids the trouble of re-allocating the python stack once python is up.

Reply:
Yea it works a-ok without heap. Big oops from me. Fixed it.